### DISCUSSION TITLE
openssl 3.2.0 does not compile with icc on windows

### DISCUSSION THREAD

**Discussion by UlrichKamke** [openssl/openssl#23412] [General Discussion]
Hello all,

this reason of compile failure is in the file openssl-3.2.0\include\internal\refcount.h line 106.

The following function seems to be wrong in parameter list:

static __inline int CRYPTO_DOWN_REF(CRYPTO_REF_COUNT *val, int *refcnt)
{
    *ret = _InterlockedExchangeAdd((void *)&refcnt->val, -1) - 1;
    return 1;
}

*ret is undefined.
Is

static __inline int CRYPTO_DOWN_REF(CRYPTO_REF_COUNT *refcnt, int *ret)
{
    *ret = _InterlockedExchangeAdd((void *)&refcnt->val, -1) - 1;
    return 1;
}

right code ?

Can someone from your team fix this?

Best regards 
Ulrich

**Comment by mattcaswell** [comment]
Created as issue #23414

**Comment by UlrichKamke** [comment]
The one fix 

static __inline int CRYPTO_DOWN_REF(CRYPTO_REF_COUNT *val, int *refcnt)

=>

static __inline int CRYPTO_DOWN_REF(CRYPTO_REF_COUNT *refcnt, int *ret) 

in line 104 of refcount.h works for us.

We replaced only this file in 3.2.0. 

Thank you

**Comment by UlrichKamke** [comment]
This issue is fixed. I replaced in our build the file refcount.h wiht the change.
Thank you.
